pkg/synthetictests/duplicated_events.go

```diff
@@ -288,31 +290,45 @@ func (d duplicateEventsEvaluator) testDuplicatedE2ENamespaceEvents(events monito
 func (d duplicateEventsEvaluator) testDuplicatedEvents(testName string, flakeOnly bool, events monitorapi.Intervals, kubeClientConfig *rest.Config) []*junitapi.JUnitTestCase {
 	allowedRepeatedEventsRegex := combinedRegexp(d.allowedRepeatedEventPatterns...)
 
-	displayToCount := map[string]int{}
+	type eventInfo struct {
+		err   error
+		count int
+	}
+	displayToEventInfo := map[string]eventInfo{}
 	for _, event := range events {
 		eventDisplayMessage, times := getTimesAnEventHappened(fmt.Sprintf("%s - %s", event.Locator, event.Message))
 		if times > duplicateEventThreshold {
 			if allowedRepeatedEventsRegex.MatchString(eventDisplayMessage) {
 				continue
 			}
+			eInfo := eventInfo{count: times}
 			allowed := false
 			for _, allowRepeatedEventFn := range d.allowedRepeatedEventFns {
-				if allowRepeatedEventFn(event, kubeClientConfig) {
-					allowed = true
+				eInfo.err, allowed = allowRepeatedEventFn(event, kubeClientConfig)
```

Comment:
could just just define `var failures []string` above the for loop and when `allowRepeatedEventFn` had an err, simple do `failures = append(failures, err.Error)`?